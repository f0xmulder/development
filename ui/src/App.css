.App {}

.App__TopBar {
    padding: 40px;
    display: flex;
    align-items: center;
}

.App__TopBar .title {
    font-size: 32px;
    font-weight: bold;
}

.App__TopBar .Navigation {
    flex: 1;
    text-align: right;
}
