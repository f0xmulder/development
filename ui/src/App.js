import React from 'react'
import { BrowserRouter as Router, Route, Link } from 'react-router-dom'

import Home from './pages/Home'
import SubmitAPI from './pages/SubmitAPI'

import './App.css'

const App = () =>
    <div className="App">
        <Router>
            <div>
                <Route exact path="/" component={Home} />
                <Route path="/submit-api" component={SubmitAPI} />

                <hr />

                <ul>
                    <li><Link to="/">Home</Link></li>
                    <li><Link to="/submit-api">Submit your API</Link></li>
                </ul>
            </div>
        </Router>
    </div>

export default App
