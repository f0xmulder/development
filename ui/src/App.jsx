// Copyright Â© VNG Realisatie 2020
// Licensed under the EUPL
//
import React, { Suspense, lazy } from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { GlobalStyles as DSGlobalStyles } from '@commonground/design-system'

import theme from './theme'
import Home from './pages/Home/Home'
import SubmitAPI from './pages/SubmitAPI/SubmitAPI'
import SubmitCode from './pages/SubmitCode/SubmitCode'
import SubmitEvent from './pages/SubmitEvent/SubmitEvent'
import About from './pages/About/About'
import APIDetail from './pages/APIDetail/APIDetail'
import EventOverview from './pages/EventOverview/EventOverview'
import APIOverview from './pages/APIOverview/APIOverview'
import CodeOverview from './pages/CodeOverview/CodeOverview'
import Header from './components/Header/Header'
import Feedback from './components/Feedback/Feedback'
import Footer from './components/Footer/Footer'
import GlobalStyles from './components/GlobalStyles/GlobalStyles'

import { AppContainer, ContentWrap } from './App.styles'

const APISpecification = lazy(() =>
  import(
    /* webpackChunkName: "api-specification" */ './pages/APISpecification/APISpecification'
  ),
)

const App = () => (
  <ThemeProvider theme={theme}>
    <AppContainer className="App">
      <DSGlobalStyles />
      <GlobalStyles />

      <ContentWrap>
        <Router>
          <Header />
          <Suspense fallback={<div />}>
            <main role="main">
              <Switch>
                <Route path="/" exact component={Home} />
                <Route path="/detail/:id" exact component={APIDetail} />
                <Route
                  path="/detail/:id/:environment/specification"
                  component={APISpecification}
                />
                <Route path="/apis" exact component={APIOverview} />
                <Route path="/apis/add" component={SubmitAPI} />
                <Route path="/apis/:id" component={APIDetail} />
                <Route path="/add-api" component={SubmitAPI} />
                <Route path="/events" exact component={EventOverview} />
                <Route path="/events/add" component={SubmitEvent} />
                <Route path="/code" exact component={CodeOverview} />
                <Route path="/code/add" component={SubmitCode} />
                <Route path="/about" component={About} />
              </Switch>
              <Feedback />
            </main>
          </Suspense>
        </Router>
      </ContentWrap>
      <Footer />
    </AppContainer>
  </ThemeProvider>
)

export default App
