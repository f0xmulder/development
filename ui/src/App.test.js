import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

describe('the navigation', () => {
  let navigation

  beforeAll(() => {
    const wrapper = shallow(<App/>)
    navigation = wrapper.find('ul')
  })

  it('should contain a link to the Homepage', () => {
    const homeLink = navigation.childAt(0).find('Link')
    expect(homeLink.props().children).toBe('Home')
    expect(homeLink.props().to).toBe('/')
  })

  it('should contain a link to the Search page', () => {
    const homeLink = navigation.childAt(1).find('Link')
    expect(homeLink.props().children).toBe('Search')
    expect(homeLink.props().to).toBe('/search')
  })

  it('should contain a link to the Submit API page', () => {
    const homeLink = navigation.childAt(2).find('Link')
    expect(homeLink.props().children).toBe('Submit your API')
    expect(homeLink.props().to).toBe('/submit-api')
  })

  it('should contain a link to the About page', () => {
    const homeLink = navigation.childAt(3).find('Link')
    expect(homeLink.props().children).toBe('About')
    expect(homeLink.props().to).toBe('/about')
  })
})
