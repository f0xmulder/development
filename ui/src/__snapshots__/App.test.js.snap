// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contains the page title 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <div>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/submit-api"
      />
      <hr />
      <ul>
        <li>
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/submit-api"
          >
            Submit your API
          </Link>
        </li>
      </ul>
    </div>
  </BrowserRouter>
</div>
`;
